regulations/static/components/Collapsible.vue

```diff
@@ -40,54 +44,61 @@ export default {
             required: false,
             default: "1s",
         },
-        direction: { //horizontal or vertical
+        direction: {
+            //horizontal or vertical
             type: String,
             required: true,
         },
     },
 
-    data: function() {
+    data: function () {
         return {
             size: 0,
             visible: true,
             isVertical: true,
             styles: {
                 overflow: "hidden",
                 transition: this.transition,
-            }
-        }
+            },
+        };
     },
 
     computed: {
-        sizeStyle: function() {
-            return this.isVertical ? 
-                { height: this.visible ? this.size : 0 } :
-                { width: this.visible ? this.size : 0 };
-        }
+        sizeStyle: function () {
+            return this.isVertical
+                ? { height: this.visible ? this.size : 0 }
+                : { width: this.visible ? this.size : 0 };
+        },
     },
 
     methods: {
-        resize: function(e) {
+        resize: function (e) {
             this.computeSize();
         },
-        toggle: function(target) {
-            if(this.name === target) {
-                this.visible = !this.visible;
+        toggle: function (target) {
+            if (this.name === target) {
+                if (!this.visible) {
+                    this.computeSize();
```

Comment:
Could this be done regardless of `visibility`?

Reply:
it can.  I have it dependent on `visibility === false` to reduce the number of times it is called - this way it is not computing size when closing, only when opening.  But that may be premature optimization.

Reply:
Right, because it doesn't need to compute the current height when closing, it's just setting the height to 0. OK makes sense to me!